src/model/application/component.rs

```diff
@@ -19,10 +20,258 @@ enum_number! {
+    pub width: Option<NonMaxU32>,
+    /// The height of the media item.
+    pub height: Option<NonMaxU32>,
+    /// The content type of the media item.
+    pub content_type: Option<FixedString>,
+}
+
+/// A component that allows you to add text to your message, similiar to the `content` field of a
+/// message.
+///
+/// [Discord docs](https://discord.com/developers/docs/components/reference#text-display)
+#[cfg_attr(feature = "typesize", derive(typesize::derive::TypeSize))]
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[non_exhaustive]
+pub struct TextDisplay {
+    /// Always [`ComponentType::TextDisplay`]
+    #[serde(rename = "type")]
+    pub kind: ComponentType,
+    /// The content of this text display component.
+    pub content: FixedString<u16>,
+}
+
+/// A Media Gallery is a component that allows you to display media attachments in an organized
+/// gallery format.
+///
+/// [Discord docs](https://discord.com/developers/docs/components/reference#media-gallery)
+#[cfg_attr(feature = "typesize", derive(typesize::derive::TypeSize))]
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[non_exhaustive]
+pub struct MediaGallery {
+    /// Always [`ComponentType::MediaGallery`]
+    #[serde(rename = "type")]
+    pub kind: ComponentType,
+    /// Array of images this media gallery can contain, max of 10.
+    pub items: FixedArray<MediaGalleryItem>,
+}
+
+/// An individual media gallery item.
+///
+/// Belongs to [`MediaGallery`].
+///
+/// [Discord docs](https://discord.com/developers/docs/components/reference#media-gallery-media-gallery-item-structure)
+#[cfg_attr(feature = "typesize", derive(typesize::derive::TypeSize))]
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[non_exhaustive]
+pub struct MediaGalleryItem {
+    /// The internal media piece that this item contains.
+    pub media: UnfurledMediaItem,
+    /// The description of the media item.
+    pub description: Option<FixedString<u16>>,
+    /// Whether or not this component is spoilered.
+    pub spoiler: Option<bool>,
+}
+
+/// A component that adds vertical padding and visual division between other components.
+///
+/// [Discord docs](https://discord.com/developers/docs/components/reference#separator)
+#[cfg_attr(feature = "typesize", derive(typesize::derive::TypeSize))]
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[non_exhaustive]
+pub struct Separator {
+    /// Always [`ComponentType::Separator`]
+    #[serde(rename = "type")]
+    pub kind: ComponentType,
+    /// Whether or not this contains a separating divider.
+    pub divider: Option<bool>,
+    /// The spacing of the separator.
+    pub spacing: Option<SeparatorSpacingSize>,
+}
+
+enum_number! {
+    /// The size of a separator component.
+    #[derive(Clone, Copy, Debug, Eq, Hash, Ord, PartialEq, PartialOrd, Deserialize, Serialize)]
+    #[cfg_attr(feature = "typesize", derive(typesize::derive::TypeSize))]
+    #[non_exhaustive]
+    pub enum SeparatorSpacingSize {
```

Comment:
This seems like it's a duplicate of the `Spacing` struct in `create_components.rs`. I don't see an immediate reason why these need to be distinct, but I may be missing something.